sxt/field_mtg/operation/add.h

```diff
@@ -28,27 +28,23 @@
 #include <cstdint>
 
 #include "sxt/base/field/arithmetic_utility.h"
+#include "sxt/base/field/element.h"
 #include "sxt/base/field/subtract_p.h"
 #include "sxt/base/macro/cuda_callable.h"
-#include "sxt/field12/base/constants.h"
-#include "sxt/field12/type/element.h"
 
-namespace sxt::f12o {
+namespace sxt::fmtgo {
 //--------------------------------------------------------------------------------------------------
 // add
 //--------------------------------------------------------------------------------------------------
-CUDA_CALLABLE inline void add(f12t::element& h, const f12t::element& f,
-                              const f12t::element& g) noexcept {
-  uint64_t h_tmp[6] = {};
+template <basfld::element Element>
+CUDA_CALLABLE inline void add(Element& h, const Element& f, const Element& g) noexcept {
```

Comment:
How is this function specific to Montgomery form?

Reply:
`add` assumes the elements it operates on are in Montgomery form. While this `add` might work for `field51` elements, ignoring the `subtract_p` call, as far as I know future operations refactored from the `field12/operation` package won't be guaranteed to work. When converting to Montgomery form care is taken to select a `R` factor, which you can be seen used in the `f12t::from_bytes` conversion. `R` keeps the same number of limbs as the non-Montgomery form base field element. The current `f51t::from_bytes` employs a different conversion, I assume to take advantage of the optimization choices made by the libsodium project, but I have not studied that project closely.

Arkworks also organizes a similar add function in `montgomery_backend.rs`. https://github.com/arkworks-rs/algebra/blob/b7b988f0aa323d20ab704c8bb1a681f215bf9bd6/ff/src/fields/models/fp/montgomery_backend.rs#L98-L107C6